### PR TITLE
[s]Fixes lava staff being used through walls

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
I mean people aready know so maybe no point in the security tag but still, no more using mesons to kill stuff through walls.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm [hunk 1]
```diff
@@ -382,7 +382,8 @@
 	..()
 	if(lava_cooldown > world.time)
 		return
-	if(target && istype(target, /turf/open))
+	var/turf/T = get_turf(user)
+	if(target && istype(target, /turf/open) && target in view(7,T))
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm]: wrap the ```target in view(7,T)``` in parenthesis, so:
```... && (target in view(7,T))```

also use user.client.view, not 7, for future proofing.

**Comment by RemieRichards** [comment]
>half reading my comment

cmon kor

File: code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm [hunk 2]
```diff
@@ -383,7 +383,7 @@
 	if(lava_cooldown > world.time)
 		return
 	var/turf/T = get_turf(user)
-	if(target && istype(target, /turf/open) && target in view(7,T))
+	if(target && istype(target, /turf/open) && (target in view(7,T)))
```
Comment (lordpidey) [inline code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm]: shouldn't we just use (istype(target, /turf/open) && (target in view(7,T))?
Isn't the first target redundant due to the istype?

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/megafauna/dragon.dm]: see:

>"half reading my comment"
cmon kor

and of course, my original comment :P

**Comment by kevinz000** [comment]
>not using it with a sniper rifle for 360 lavascope

**Pull request merged**